src/main/java/com/rocketden/main/controller/v1/BaseRestController.java

```diff
@@ -2,7 +2,11 @@
 
 import org.springframework.web.bind.annotation.RequestMapping;
 
+// Sets the base path for API endpoints. Custom controllers should extend this class.
 @RequestMapping("/api/v1")
 public class BaseRestController {
-    // Sets the base path for API endpoints. Custom controllers should extend this class.
+
+    // Constants to hold custom paths.
+    public static final String BASE_URL = "/api/v1";
+    public static final String BASE_SOCKET_URL = "/api/v1/socket";
```

Comment:
There is a **sonarlint** warning (see image) on this class as it sees it as a Utility class that should thus have a private constructor; however, when I added this, it created errors when running the application.

<img width="697" alt="BaseRestController Error" src="https://user-images.githubusercontent.com/7987279/92516310-df949a80-f1c9-11ea-845c-48c772d97745.png">

Thus, I didn't address the error; if there is any action I should take to address the warning, please let me know.

Reply:
Making it a `final` class is a good idea, but because other classes extend `BaseRestController` (see `UserSocketController extends BaseRestController`), so if this class was `final` then the rest of the classes would have to be `final` as well (which is not correct).

As for the constructor, I think you're right that an empty constructor actually works great! I didn't realize this the first time because I had included an error that is thrown in the constructor as the **sonarlint** description / example had shown, but this prevented `./mvnw spring-boot:run` from running properly. I had to make the constructor `protected`, as otherwise the classes that extended `BaseRestController` threw the following error:
<img width="537" alt="Private Constructor Error" src="https://user-images.githubusercontent.com/7987279/92740645-c2c7a680-f332-11ea-8518-362274713fbd.png">
All in all, I added the lines:
```
/** 
  * Protected empty constructor to allow visability from extended classes,
  * but also remove warning on Utility class having no constructor to hide
  * the implicit public one.
  */ 
protected BaseRestController() {}
```
Thanks for the comment!